CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua

```diff
@@ -0,0 +1,59 @@
+local CurveUtil = {	}
+
+function CurveUtil:Expt(start, to, pct, dt_scale)
+	if math.abs(to - start) < 0.05 then
+		return to
+	end
+	
+	local y = CurveUtil:Expty(start,to,pct,dt_scale)
+			
+	--rtv = start + (to - start) * timescaled_friction--
+	local delta = (to - start) * y
+	return start + delta
+end
+
+function CurveUtil:Expty(start, to, pct, dt_scale)	
+	--y = e ^ (-a * timescale)--
+	local friction = 1 - pct
+	local a = -math.log(friction)
+	return 1 - math.exp(-a * dt_scale)
+end
+
+function CurveUtil:BezierValForT(p0, p1, p2, p3, t)
```

Comment:
This seems like a fairly computationally expensive transformation, is this something that is used much?